src/parser/vast_parser.js

```diff
@@ -263,9 +295,74 @@ export class VASTParser extends EventEmitter {
         lastAddedAd.sequence = wrapperSequence;
       }
 
+      this.resolveAds(ads, { resolveAll, wrapperDepth, originalUrl })
+        .then(res => resolve(res))
+        .catch(err => reject(err));
+    });
+  }
+
+  /**
+   * Resolves an Array of ads, recursively calling itself with the reamining ads if a no ad
```

Comment:
remaining ads ? typo ?

Reply:
Is this a valid algorithm to split sequences :

```javascript
var ads = [
	{
		id: 1
	},
	{
		id: 2,
		sequence: 1
	},
	{
		id: 3,
		sequence: 2
	},
	{
		id: 4
	},
	{
		id: 5,
		sequence: 1
	},
	{
		id: 6,
		sequence: 2
	},
	{
		id: 7,
		sequence: 3
	},
	{
		id: 8,
		sequence: 1
	},
	{
		id: 9,
		sequence: 2
	},
	{
		id: 10
	},
	{
		id: 11,
		sequence: 1
	},
	{
		id: 12
	}
]

var pods = []

ads.forEach((ad) => {
  if (ad.sequence && ad.sequence > 1) {
    pods[pods.length - 1].push(ad)
    return
  }
  pods.push([ad])
})

console.log(pods)
```